torch/_inductor/decomposition.py

```diff
@@ -591,6 +591,37 @@ def get_like_layout(
         return memory_format
 
 
+def _preserve_stride_pattern(
+    result: torch.Tensor,
+    reference: torch.Tensor,
+    memory_format: Optional[torch.memory_format] = None,
+) -> torch.Tensor:
+    """
+    Preserve the stride pattern of the reference tensor in the result tensor.
+
+    This is critical for maintaining the correct memory layout, especially for
+    non-contiguous tensors (e.g., after rot90).
+
+    Args:
+        result: The tensor to adjust
+        reference: The tensor whose stride pattern to match
+        memory_format: Optional memory format to apply
+
+    Returns:
+        The result tensor with matching stride pattern
+    """
+    # If memory_format is explicitly set, use it
+    if memory_format is not None and memory_format != torch.preserve_format:
+        return result.to(memory_format=memory_format)
```

Comment:
The documentation is a bit unclear that the stride will not match if the memory_format is specified. What if I want channels_last, channels_first memory format with a specific stride? 

I guess I don't know if the invariant that non preserve memory format always implies standard striding is true.

Reply:
this is the original logic that orders when to preserve memory formats etc. https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/TensorFactories.cpp#L461-L489